projects/packages/publicize/src/class-social-admin-page.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Social Admin Page class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize;
+
+use Automattic\Jetpack\Admin_UI\Admin_Menu;
+use Automattic\Jetpack\Assets;
+
+/**
+ * The class to handle the Social Admin Page.
+ */
+class Social_Admin_Page {
+
+	/**
+	 * The instance of the class.
+	 *
+	 * @var Social_Admin_Page
+	 */
+	private static $instance;
+
+	/**
+	 * Initialize the class.
+	 *
+	 * @return Social_Admin_Page
+	 */
+	public static function init() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * The constructor.
+	 */
+	private function __construct() {
+		add_action( 'admin_menu', array( $this, 'add_menu' ) );
+
+		/**
+		 * Admin_Menu::add_menu uses 1000, so we use 2000
+		 * to ensure we remove the old Social menu item after it has been added.
+		 */
+		add_action( 'admin_menu', array( $this, 'remove_old_social_menu_item' ), 2000 );
+	}
+
+	/**
+	 * Remove the page added by old versions of Social.
+	 */
+	public function remove_old_social_menu_item() {
+
+		if ( defined( 'JETPACK_SOCIAL_PLUGIN_ROOT_FILE' ) ) {
+
+			$plugin_data = get_plugin_data( (string) constant( 'JETPACK_SOCIAL_PLUGIN_ROOT_FILE' ), false, false );
+
+			$plugin_version = $plugin_data['Version'];
+
+			// If it's the old social version, remove the submenu page.
+			// TODO Update the version and operator before next Social release.
+			if ( version_compare( $plugin_version, '6.1.0', '<' ) ) {
+				/**
+				 * `add_submenu_page` allows multiple submenus with the same slug,
+				 * but `remove_submenu_page` only removes the first one it finds with the given slug.
+				 *
+				 * We add the menu using `admin_menu` hook unlike the old Social plugin,
+				 * which used the `init` hook, which runs before `admin_menu`.
+				 * So, the old Social plugin's menu is before the new one in $submenu global.
+				 *
+				 * So, `remove_submenu_page` should remove the menu added by the old Social plugin.
+				 *
+				 * @see https://developer.wordpress.org/reference/functions/add_submenu_page
+				 * @see https://developer.wordpress.org/reference/functions/remove_submenu_page
+				 */
+				remove_submenu_page( 'jetpack', 'jetpack-social' );
+			}
+		}
+	}
+
+	/**
+	 * Add the admin menu.
+	 */
+	public function add_menu() {
+
+		// If Social plugin is active.
+		if ( defined( 'JETPACK_SOCIAL_PLUGIN_DIR' ) ) {
```

Comment:
I'm not sure why, but there's something about checking this constant which doesn't feel right for this purpose. Is there a better way to check for this? I suppose we could do a check for the `Jetpack_Social` class. I know that `is_plugin_active` is problematic too. It's not a blocker, but something to think about.

Reply:
This is temporary until we enable the page for all the sites, you can see it already being removed in #41278

Reply:
Ok! If it's shortlived, that's fine.